Fix: Update Jest config and package dependencies

timDeHof opened the pull request:
# Description

<!-- Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change. -->
Fixed Jest configuration and dependency issues to improve development workflow and CI/CD pipeline.

- Updated Jest config to use proper async configuration setup
- Fixed @chingu-x/components dependency from local file path to npm version
- Removed Cypress from pre-push hook to prevent local test failures during branch pushes

## Issue link

Fixes # (issue)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

- Verified Jest tests run successfully with new configuration
- Confirmed git push works without Cypress errors
- Tested that semantic-release only runs on main branch

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

timDeHof committed:
chore(package): implement code changes to enhance functionality and improve performance

timDeHof committed:
chore: add code owner for timDeHof

timDeHof committed:
refactor(ui): improve string splitting logic for voyage team display

timDeHof committed:
refactor(MeetingForm): ✏️ improve code formatting and indentation

* Updated indentation from 2 to 4 spaces in ESLint configuration.
* Refactored object structure for clarity in `MeetingForm` component.
* Ensured consistent formatting for better readability.

timDeHof committed:
refactor(MeetingForm): ✏️ improve code formatting for clarity

* Adjusted indentation for better readability in the `onSubmit` function.
* Standardized formatting in the conditional object construction for `meetingLink`.
* Enhanced consistency in the comparison logic within the `useEffect` hook.

timDeHof committed:
refactor(eslint): ✏️ remove unnecessary languageOptions for file overrides

* Cleaned up ESLint configuration by removing redundant languageOptions.
* This simplifies the configuration and improves maintainability.

timDeHof committed:
refactor(eslint): ✏️ change indentation from 4 to 2 spaces

* Updated the "indent" rule in ESLint configuration to enforce 2 spaces for indentation.
* This change improves consistency with the project's coding style.

timDeHof committed:
refactor(MeetingForm): improve code formatting and clarity

timDeHof committed:
refactor(MeetingForm): ✏️ improve code formatting for clarity

* Adjusted indentation for better readability in the MeetingForm component.
* Ensured consistent formatting in the object structure for meeting data.

timDeHof committed:
refactor(MeetingForm): ✏️ improve code formatting for consistency

* Adjusted indentation for better readability.
* Ensured consistent formatting across the `MeetingForm` component.

timDeHof committed:
refactor(MeetingForm): ✏️ improve code formatting and consistency

* Adjusted indentation for better readability.
* Ensured consistent formatting across the `MeetingForm` component.
* Organized imports and function definitions for clarity.

timDeHof committed:
refactor(DateTimePicker): ✏️ update `react-datepicker` to version 8.7.0

* Changed import from `react-datepicker` to `ReactDatePicker`.
* Removed unnecessary extension of `ReactDatePickerProps` in `DatePickerInputProps`.
* Refactored `DateTimePicker` to a standard function component.

timDeHof committed:
refactor(DateTimePicker): ✏️ improve code formatting for consistency

* Adjusted indentation for better readability.
* Ensured consistent formatting across the component.

timDeHof committed:
refactor(MeetingForm): ✏️ update `setCustomValue` to handle null values

* Modified `setCustomValue` to accept `Date | null` instead of just `Date`.
* Added a conditional check to ensure `setValue` is only called with valid dates.
* Updated the `onChange` handler to reflect the new type for `dateTime`.

timDeHof committed:
refactor(jest.config): ✏️ remove unnecessary `await` from `createJestConfig`

* Simplified the configuration by removing `await` as `createJestConfig` is not an async function.

timDeHof committed:
refactor: restructure imports for better type clarity

timDeHof committed:
refactor(Card.test): ✏️ improve code formatting for consistency

* Adjusted indentation for better readability.
* Ensured consistent formatting across mock implementations.

timDeHof committed:
refactor(Card.test): ✏️ restructure imports for better clarity

* Combined import statements for `Feature` and `features` from `./fixtures/Features`.
* Improved code formatting for consistency.

timDeHof committed:
refactor(Card.test): ✏️ update mock for `EditPopover` to use simpler string representation

* Simplified the mock implementation of `EditPopover` for clarity.
* This change enhances readability and maintains consistency in mock definitions.

timDeHof committed:
chore(config): add babel config, update jest, and enhance tests

- Added babel.config.js for better TypeScript and React support
- Updated jest.config.mjs to use createJestConfig
- Added babel dependencies to package.json
- Enhanced test setup with improved mocks

timDeHof committed:
feat(testing): add SWC for Jest transpilation

- Update Jest config to use @swc/jest transformer
- Add @swc/core and @swc/jest dependencies
- Improve test transpilation performance with SWC

timDeHof committed:
feat: upgrade cypress to v13.0.0 and its dependencies

- Update cypress from 12.17.3 to 13.0.0
- Update several cypress dependencies to compatible versions

timDeHof committed:
test(test): skip sign in test until valid credentials are available

timDeHof committed:
ci: add branch condition to release job and optimize checkout process

- added `if: github.ref == 'refs/heads/main'` to ensure job runs only on main branch
- optimized checkout step with `fetch-depth: 0` for faster process
- removed `yarn cypress` from husky pre-push hook

cherylli committed:
change vercel to use yarn 1

cherylli committed:
update gitignore

cherylli committed:
delete .idea folder

cherylli commented on .github/CODEOWNERS:
I think we can just remove codeowners as discussed on discord

timDeHof replied:
I have removed codeowners

cherylli commented:
> Removed Cypress from pre-push hook to prevent local test failures during branch pushes

why would cypress fail? is this the expected behaviour? shouldn't we make sure cypress tests pass before pushing?

cherylli commented on src/app/(main)/my-voyage/[teamId]/features/components/Card.test.tsx:
```diff
@@ -3,108 +3,100 @@ import { configureStore } from "@reduxjs/toolkit";
 import Card from "./Card";
-import { type Feature, features } from "./fixtures/Features";
+// eslint-disable-next-line @typescript-eslint/consistent-type-imports
+import { Feature, features } from "./fixtures/Features";
```
we should probably leave the `type` here and remove the eslint disable comment

timDeHof replied:
I fixed this by importing the 'Feature' type separately

cherylli reviewed (commented):
Are these files just formatting changes?
`src/components/navbar/UserDropdown.tsx`
`src/app/(main)/my-voyage/[teamId]/features/components/Card.test.tsx`

timDeHof committed:
docs(github): add pull request template

- Moved pull request template to .github folder
- Removed code owners file
- Updated husky pre-push hook
- Refactored test file imports and test cases

timDeHof committed:
fix(dependencies): 🔧 update `cypress` to version 15.2.0

* Upgraded `cypress` from `^13.0.0` to `^15.2.0` in `package.json`.

timDeHof commented:
> Are these files just formatting changes? `src/components/navbar/UserDropdown.tsx` `src/app/(main)/my-voyage/[teamId]/features/components/Card.test.tsx`

yes, there was a conflict between eslint and prettier with spaces

> why would cypress fail? is this the expected behaviour? shouldn't we make sure cypress tests pass before pushing?

The tests are passing successful when I ran the command alone. it was when I tried to sync the changes with `git push origin dev-review:dev-review`. The git logs would show the following:
![Screenshot 2025-09-17 at 10 18 56 AM](https://github.com/user-attachments/assets/a16cfa6b-3517-41ac-8e5e-bdce14cef39c)
 
 But running `git push` in the terminal worked.

cherylli commented:
> sing successful when I ran the command alone. it was when I tried to sync the changes with `git push origin dev-review:dev-review`.

I see. Sure we can keep cypress removed from the pre push hook for now and fix it in the other PR
I think husky is not working, I had a lot of issues with husky for this PR (or maybe this repo in general)

cherylli commented:
> yes, there was a conflict between eslint and prettier with spaces

We should fix this, so we don't see a lot of format changes. Makes it hard to review
